yt_dlp/extractor/dplay.py

```diff
@@ -426,3 +436,63 @@ class AnimalPlanetIE(DiscoveryPlusIE):
 
     _PRODUCT = 'apl'
     _API_URL = 'us1-prod-direct.animalplanet.com'
+
+
+class DiscoveryPlusShowBaseIE(InfoExtractor):
+    def _entries(self, show_name):
+        headers = {
+            'x-disco-client': self._X_CLIENT,
+            'x-disco-params': 'realm=' + self._REALM,
+            'referer': self._DOMAIN,
+        }
+        headers['Authorization'] = 'Bearer ' + self._download_json(
+            self._BASE_API + 'token', show_name, 'Downloading token',
+            query={
+                'realm': self._REALM,
+                'deviceId': uuid.uuid4().hex,
+            })['data']['attributes']['token']
+        show_url = self._BASE_API + 'cms/routes/' + self._SHOW_STR + '/{}?include=default'.format(show_name)
+        show_json = self._download_json(show_url,
+                                        video_id=show_name,
+                                        headers=headers)['included'][self._INDEX]['attributes']['component']
+        show_id = show_json['mandatoryParams'].split('=')[-1]
+        season_url = self._BASE_API + 'content/videos?sort=episodeNumber&filter[seasonNumber]={}&filter[show.id]={}&page[size]=100&page[number]={}'
+        for season in show_json['filters'][0]['options']:
+            season_id = season['id']
+            total_pages, page_num = 1, 0
+            while page_num < total_pages:
+                season_json = self._download_json(season_url.format(season_id, show_id, str(page_num + 1)),
+                                                  video_id=show_id, headers=headers,
+                                                  note='Downloading JSON metadata%s' % (' page %d' % page_num if page_num else ''))
+                if page_num == 0:
+                    total_pages = try_get(season_json, lambda x: x['meta']['totalPages'], int) or 1
+                episodes_json = season_json['data']
+                for episode in episodes_json:
+                    video_id = episode['attributes']['path']
+                    yield self.url_result(
+                        '%svideos/%s' % (self._DOMAIN, video_id),
+                        ie=self._VIDEO_IE.ie_key(), video_id=video_id)
+                page_num += 1
+
+    def _real_extract(self, url):
+        show_name = self._match_valid_url(url).group('show_name')
```

Comment:
might as well use `_match_id`

Reply:
if this was spanning multiple lines for some reason, then yeah sure i would have left it as it is
But it's 1 line and more importantly allows dev to extend this to parse the domain if the _INDEX is same for 2 diff regions.